src/main/java/com/blackducksoftware/integration/jira/common/BlackDuckAssignUtil.java

```diff
@@ -0,0 +1,133 @@
+
+            final PluginBlackDuckServerConfigModel blackDuckServerConfig = globalConfigurationAccessor.getBlackDuckServerConfig();
+            final BlackDuckServicesFactory blackDuckServicesFactory = blackDuckConnectionHelper.createBlackDuckServicesFactory(logger, blackDuckServerConfig.createBlackDuckServerConfigBuilder());
+
+            final BlackDuckService blackDuckService = blackDuckServicesFactory.createBlackDuckService();
+            final Set<ProjectView> matchingProjects = getMatchingBDProjects(config.getHubProjectMappings(), blackDuckService);
+
+            if (matchingProjects.isEmpty()) {
+                logger.debug("There are no BD projects that map the projects configured in the project mappings. Skipping assigning the user to the BD Project.");
+                return;
+            }
+            final UserView currentUser = blackDuckService.getResponse(ApiDiscovery.CURRENT_USER_LINK_RESPONSE);
+            final Set<String> assignedProjects = blackDuckService.getAllResponses(currentUser, UserView.PROJECTS_LINK_RESPONSE)
+                                                     .stream()
+                                                     .map(assignedProject -> assignedProject.getName())
+                                                     .collect(Collectors.toSet());
+            final Set<ProjectView> nonAssignedProjects = matchingProjects.stream()
+                                                             .filter(project -> !assignedProjects.contains(project.getName()))
+                                                             .collect(Collectors.toSet());
+
+            if (nonAssignedProjects.isEmpty()) {
+                logger.debug("There are no BD projects that need to have this User assigned to them. Skipping assigning the user to the BD Project.");
+                return;
+            }
+
+            final AssignedUserRequest assignedUserRequest = new AssignedUserRequest();
+            assignedUserRequest.setUser(currentUser.getHref().orElseThrow(() -> new IntegrationException(String.format("The current user, %s, does not have an href.", currentUser.getUserName()))));
+            for (final ProjectView projectView : nonAssignedProjects) {
+                final Optional<String> projectUsersLinkOptional = projectView.getFirstLink(ProjectView.USERS_LINK);
+                if (projectUsersLinkOptional.isPresent()) {
+                    blackDuckService.post(projectUsersLinkOptional.get(), assignedUserRequest);
+                } else {
+                    logger.error(String.format("Could not assign the user, %s, to the project %s because there is no users link.", currentUser.getUserName(), projectView.getName()));
+                }
+            }
+
+        } catch (final IntegrationException e) {
+            logger.error("Could not assign the Black Duck user to the configured Black Duck projects. " + e.getMessage(), e);
```

Comment:
I think we should bubble this up to the Global UI so the user knows why issues might not be created.